ditail.py

```diff
@@ -44,27 +46,38 @@ def tail_task(path, modify_event, loop, new=False):
                         break
                     else:
                         # Write out through newline, remove that portion from buffer
+                        sys.stdout.write(path.encode())
+                        sys.stdout.write(b":")
                         sys.stdout.write(line_buffer[:(n+1)])
                         line_buffer = line_buffer[(n+1):]
     finally:
         os.close(f)
 
-def update_tasks_in(directory, tail_tasks, tail_events, loop):
+def update_tasks_in(directory, tail_tasks, tail_events, loop, filter_regexp, disable_truncate_detect):
     wm = pyinotify.WatchManager()
     mask = pyinotify.IN_DELETE | pyinotify.IN_CREATE | pyinotify.IN_MODIFY
 
     class Handler(pyinotify.ProcessEvent):
+        def __init__(self, filter_regexp, disable_truncate_detect):
+            self.filter_regexp = filter_regexp
+            self.disable_truncate_detect = disable_truncate_detect
```

Comment:
The `Handler` class is a closure within `update_tasks_in` (there will be a new class created on every execution of `update_tasks_in`), so there's no need to set the function parameters as object variables. The class sees the local namespace of the function it was created in.